pennylane/ops/op_math/evolve.py

```diff
@@ -0,0 +1,215 @@
+
+    Args:
+        op (Union[.Operator, .ParametrizedHamiltonian]): operator to evolve
+
+    Returns:
+        Union[.Evolution, ~pennylane.ops.op_math.evolve.ParametrizedEvolution]: evolution operator
+
+    .. seealso:: :class:`.ParametrizedEvolution`
+    .. seealso:: :class:`.Evolution`
+
+    **Examples**
+
+    We can use ``qml.evolve`` to compute the evolution of any PennyLane operator:
+
+    >>> op = qml.s_prod(2, qml.PauliX(0))
+    >>> qml.evolve(op)
+    Exp((-0-1j) 2*(PauliX(wires=[0])))
+
+    When evolving a :class:`.ParametrizedHamiltonian` class, then a :class:`.ParametrizedEvolution`
+    instance is returned:
+
+    >>> coeffs = [lambda p, t: p * t for _ in range(4)]
+    >>> ops = [qml.PauliX(i) for i in range(4)]
+    >>> H = qml.ops.dot(coeffs, ops)
+    >>> qml.evolve(H)
+    ParametrizedEvolution(wires=[0, 1, 2, 3])
+
+    Please check the :class:`.ParametrizedEvolution` class for more information.
+    """
+    if isinstance(op, ParametrizedHamiltonian):
+        return ParametrizedEvolution(H=op)
+
+    return Evolution(generator=op, param=1.0)
+
+
+class ParametrizedEvolution(Operation):
+    r"""Parametrized evolution gate.
+
+    For a time-dependent Hamiltonian of the form
+    :math:`H(v, t) = H_\text{drift} + \sum_j f_j(v, t) H_j` it implements the corresponding
+    time-evolution operator :math:`U(t_1, t_2)`, which is the solution to the time-dependent
+    Schrodinger equation.
+
+    .. math:: \frac{d}{dt}U(t) = -i H(v, t) U(t).
+
+
+    Under the hood, it is using a numerical ordinary differential equation solver.
+
+    Args:
+        H (ParametrizedHamiltonian): hamiltonian to evolve
+        params (ndarray): trainable parameters
+        t (Union[float, List[float]]): If a float, it corresponds to the duration of the evolution.
+            If a list of two floats, it corresponds to the initial time and the final time of the
+            evolution. Note that such absolute times only have meaning within an instance of
+            ``ParametrizedEvolution`` and will not affect other gates.
+        time (str, optional): The name of the time-based parameter in the parametrized Hamiltonian.
+            Defaults to "t".
+        do_queue (bool): determines if the scalar product operator will be queued. Default is True.
+        id (str or None): id for the scalar product operator. Default is None.
+
+    .. warning::
+
+        The time argument ``t`` corresponds to the time window used to compute the scalar-valued
+        functions present in the :class:`ParametrizedHamiltonian` class. Consequently, executing
+        two ``ParametrizedEvolution`` gates using the same time window does not mean both gates
+        are executed simultaneously, but rather both gates evaluate their respective scalar-valued
+        functions using the same time window.
+
+    .. note::
+
+        To execute two :class:`ParametrizedHamiltonian`s simultaneously one must wrap them with the
```

Comment:
```suggestion
        To execute two :class:`.ParametrizedHamiltonian`s simultaneously one must wrap them with the
```

Reply:
Seems to render incorrectly if there is no space after the second `

Reply:
Solved in a new PR: https://github.com/PennyLaneAI/pennylane/pull/3656